core/src/main/java/org/elasticsearch/action/bulk/byscroll/DeleteByQueryRequest.java

```diff
@@ -111,4 +111,16 @@ public IndicesOptions indicesOptions() {
         assert getSearchRequest() != null;
         return getSearchRequest().indicesOptions();
     }
+
+    public String[] types() {
+        assert getSearchRequest() != null;
+        return getSearchRequest().types();
+    }
+
+    public DeleteByQueryRequest types(String... types) {
```

Comment:
sorry @kunal642 one last thing, could you also test this method please? set the types using this new method and check that the underlying search request types reflect the change?